### PR TITLE
[HERA-354] Onboarding pages admin side [LMS]

### PR DESCRIPTION
**Pull request by drtwisted**
[HERA-354](https://youtrack.raccoongang.com/issue/HERA-354) Onboarding pages admin side [LMS]
- create the defult Onboarding model instance if there is none
- prohibit adding new Onboarding instances in case there are any present
- prohibit deleting Onboarding instances

### CODE REVIEW

File: lms/djangoapps/hera/admin.py [hunk 1]
```diff
@@ -13,7 +13,29 @@ class OnboardingAdmin(admin.ModelAdmin):
...
+        if not qs:
+            default_onboarding = self.model()
+            default_onboarding.save()
+
+            qs = super(OnboardingAdmin, self).get_queryset(request)
+
+        return qs
+
+    def has_add_permission(self, request):
+        return self.model.objects.all().count() == 0
```
Comment (cmltaWt0) [inline lms/djangoapps/hera/admin.py]:
```suggestion
        return not self.model.objects.exists()
```

File: lms/djangoapps/hera/admin.py [hunk 2]
```diff
@@ -13,7 +13,29 @@ class OnboardingAdmin(admin.ModelAdmin):
...
+
+        if not qs:
+            default_onboarding = self.model()
+            default_onboarding.save()
+
+            qs = super(OnboardingAdmin, self).get_queryset(request)
+
+        return qs
+
+    def has_add_permission(self, request):
```
Comment (cmltaWt0) [inline lms/djangoapps/hera/admin.py]: Need to add docstring.
For a reviewer not obvious how `queryset.count` relates with `has_add_permissios` flag.

File: lms/djangoapps/hera/admin.py [hunk 3]
```diff
@@ -13,7 +13,29 @@ class OnboardingAdmin(admin.ModelAdmin):
     """
     Reflection of the hera model on the admin panel.
     """
-    pass
+
+    def get_queryset(self, request):
+        qs = super(OnboardingAdmin, self).get_queryset(request)
```
Comment (cmltaWt0) [inline lms/djangoapps/hera/admin.py]: Also need to add docstring for this case.

File: lms/djangoapps/hera/admin.py [hunk 4]
```diff
@@ -13,7 +13,29 @@ class OnboardingAdmin(admin.ModelAdmin):
...
+
+        if not qs:
```
Comment (cmltaWt0) [inline lms/djangoapps/hera/admin.py]: Have you considered to use decorator for this?

File: lms/djangoapps/hera/admin.py [hunk 5]
```diff
@@ -13,7 +13,29 @@ class OnboardingAdmin(admin.ModelAdmin):
...
+
+            qs = super(OnboardingAdmin, self).get_queryset(request)
+
+        return qs
+
+    def has_add_permission(self, request):
+        return self.model.objects.all().count() == 0
+
+    def get_actions(self, request):
+        delete_action = 'delete_selected'
```
Comment (cmltaWt0) [inline lms/djangoapps/hera/admin.py]: Docstring. Completely not obvious what is happening here.

File: lms/djangoapps/hera/admin.py (hunk 2 above)
Comment (drtwisted) [reply lms/djangoapps/hera/admin.py]: Done.

File: lms/djangoapps/hera/admin.py (hunk 1 above)
Comment (drtwisted) [reply lms/djangoapps/hera/admin.py]: Done.

File: lms/djangoapps/hera/admin.py (hunk 5 above)
Comment (drtwisted) [reply lms/djangoapps/hera/admin.py]: Done.

File: lms/djangoapps/hera/admin.py (hunk 4 above)
Comment (drtwisted) [reply lms/djangoapps/hera/admin.py]: @cmltaWt0 I quit can't see why and where a decorator is applicable here. Could you provide more details?

File: lms/djangoapps/hera/admin.py (hunk 3 above)
Comment (drtwisted) [reply lms/djangoapps/hera/admin.py]: Done.

File: lms/djangoapps/hera/admin.py (hunk 4 above)
Comment (cmltaWt0) [reply lms/djangoapps/hera/admin.py]: Decorator to create `default_onboarding` in a different way.
Something like:
```python
@ensure_default_exists
def def get_queryset(self, request):
    return super(OnboardingAdmin, self).get_queryset(request)
```


It's just a thoughts :) Don't worry.

Comment (drtwisted) [reply lms/djangoapps/hera/admin.py]: ok, got it.

File: lms/djangoapps/hera/admin.py [hunk 6]
```diff
@@ -13,7 +13,38 @@ class OnboardingAdmin(admin.ModelAdmin):
...
+        """
+        Allow to add Onboarding only if no objects exist.
+        """
+        return not self.model.objects.exists()
+
+    def get_actions(self, request):
+        """
+        Remove 'delete_selected' action from the list of available actions.
+        """
+        delete_action = 'delete_selected'
```
Comment (sendr) [inline lms/djangoapps/hera/admin.py]: maybe it worths returning an empty list, don't think we will be adding any actions here.
```suggestion
        return []
```

Comment (drtwisted) [reply lms/djangoapps/hera/admin.py]: Ok, will remove it.

Comment (cmltaWt0) [reply lms/djangoapps/hera/admin.py]: Not get it. But it is because I am not in the context.

File: lms/djangoapps/hera/admin.py [hunk 7]
```diff
@@ -13,7 +13,32 @@ class OnboardingAdmin(admin.ModelAdmin):
...
+        """
+        Allow to add Onboarding only if no objects exist.
+        """
+        return not self.model.objects.exists()
+
+    def get_actions(self, request):
+        """
+        Remove 'delete_selected' action from the list of available actions.
+        """
+        return {}
```
Comment (cmltaWt0) [inline lms/djangoapps/hera/admin.py]: Is it really correct?

File: lms/djangoapps/hera/admin.py [hunk 8]
```diff
@@ -13,7 +13,32 @@ class OnboardingAdmin(admin.ModelAdmin):
...
+
+    def has_add_permission(self, request):
+        """
+        Allow to add Onboarding only if no objects exist.
+        """
+        return not self.model.objects.exists()
+
+    def get_actions(self, request):
+        """
+        Remove 'delete_selected' action from the list of available actions.
```
Comment (cmltaWt0) [inline lms/djangoapps/hera/admin.py]: Is this docstring correct?

Comment (cmltaWt0) [reply lms/djangoapps/hera/admin.py]: @drtwisted @sendr

File: lms/djangoapps/hera/admin.py (hunk 7 above)
Comment (cmltaWt0) [reply lms/djangoapps/hera/admin.py]: @drtwisted @sendr

Comment (sendr) [reply lms/djangoapps/hera/admin.py]: Yes, there was an OrderedDict, I was wrong with my comment above.

Comment (cmltaWt0) [reply lms/djangoapps/hera/admin.py]: Should we change the docsting?

Comment (cmltaWt0) [reply lms/djangoapps/hera/admin.py]: @drtwisted please do it.

Comment (drtwisted) [reply lms/djangoapps/hera/admin.py]: done.

File: lms/djangoapps/hera/admin.py [hunk 9]
```diff
@@ -28,6 +28,31 @@ class Media:
...
+
+    def has_add_permission(self, request):
+        """
+        Allow to add Onboarding only if no objects exist.
+        """
+        return not self.model.objects.exists()
+
+    def get_actions(self, request):
+        """
+        Remove all acitons by returning an empty dictionary.
```
Comment (cmltaWt0) [inline lms/djangoapps/hera/admin.py]: Typo :)

Comment (cmltaWt0) [reply lms/djangoapps/hera/admin.py]: `acitons => actions`

Comment (drtwisted) [reply lms/djangoapps/hera/admin.py]: done.

**Pull request merged**